src/Stratis.Bitcoin.Tests/P2P/PeerConnectorTests.cs

```diff
@@ -210,6 +210,168 @@ public void PeerConnectorDiscovery_ConnectsTo_DiscoveredPeers()
+
+            IConnectionManager connectionManagerExisting = this.CreateConnectionManager(nodeSettings, connectionManagerSettingsExisting, peerAddressManager, peerConnector);
+            peerConnector.Initialize(connectionManagerExisting);
+
+            // Peer 1 is a local address.
+            IPAddress originalAddressPeer1 = IPAddress.Parse("::ffff:192.168.0.1"); // ipv4
+            var endpointPeer1 = new IPEndPoint(originalAddressPeer1, 80);
+            bool connectedToThisPeer = ConnectToPeer(peerAddressManager, networkPeerFactoryExisting, connectionManagerSettingsExisting, peerConnector, endpointPeer1);
+            Assert.True(connectedToThisPeer);
+
+            // Peer 2 is a local address in a different group.
+            IPAddress addressInDifferentNetworkGroupPeer2 = IPAddress.Parse("::ffff:192.168.1.1"); // ipv4
+            var endpointPeer2 = new IPEndPoint(addressInDifferentNetworkGroupPeer2, 80);
+            connectedToThisPeer = ConnectToPeer(peerAddressManager, networkPeerFactoryExisting, connectionManagerSettingsExisting, peerConnector, endpointPeer2);
+            Assert.True(connectedToThisPeer);
+
+            // Peer 3 is a local address in a different group.
+            IPAddress addressInDifferentNetworkGroupPeer3 = IPAddress.Parse("0:0:0:0:0:ffff:c0a8:101"); // ipv6
+            var endpointPeer3 = new IPEndPoint(addressInDifferentNetworkGroupPeer3, 80);
+            connectedToThisPeer = ConnectToPeer(peerAddressManager, networkPeerFactoryExisting, connectionManagerSettingsExisting, peerConnector, endpointPeer3);
+            Assert.True(connectedToThisPeer);
+        }
+
+        private static bool ConnectToPeer(PeerAddressManager peerAddressManager, Mock<INetworkPeerFactory> networkPeerFactoryExisting, ConnectionManagerSettings connectionManagerSettingsExisting, PeerConnectorConnectNode peerConnector, IPEndPoint endpointNode)
+        {
+            peerAddressManager.AddPeer(endpointNode, IPAddress.Loopback);
+            Mock<INetworkPeer> networkPeer = new Mock<INetworkPeer>();
+            networkPeer.SetupGet(np => np.PeerEndPoint).Returns(new IPEndPoint(endpointNode.Address, endpointNode.Port));
+            networkPeer.SetupGet(np => np.RemoteSocketAddress).Returns(endpointNode.Address);
+            networkPeer.SetupGet(np => np.RemoteSocketPort).Returns(endpointNode.Port);
+            networkPeer.SetupGet(np => np.State).Returns(NetworkPeerState.HandShaked);
+            networkPeerFactoryExisting.Setup(npf =>
+                npf.CreateConnectedNetworkPeerAsync(It.Is<IPEndPoint>(x => Equals(x, endpointNode)), 
+                    It.IsAny<NetworkPeerConnectionParameters>(), It.IsAny<NetworkPeerDisposer>())).Returns(Task.FromResult(networkPeer.Object));
+            connectionManagerSettingsExisting.AddNode.Add(endpointNode);
+            connectionManagerSettingsExisting.Connect.Add(endpointNode);
+            AddMissingPeersToConnectionManager(peerConnector);
```

Comment:
@thefazzer here you can just use mock a new instance of ConnectionManager and to its collection of ConnectedPeers and the initialize the PeerConnector with it :)

Reply:
This will eliminate the need for reflection etcetc

Reply:
Done.